base/iterators.jl

```diff
@@ -1337,24 +1337,40 @@ julia> sum(a) # Sum the remaining elements
 7
 ```
 """
-mutable struct Stateful{T, VS}
+mutable struct Stateful{T, VS, N <: Signed}
```

Comment:
```suggestion
mutable struct Stateful{T, VS, N<:Integer}
```
This should not need to assume it is `signed`

Reply:
Ah right. If their length implementation returns an unsigned, this never underflows anyway.